data/pigui/modules/flight-ui/target-scanner.lua

```diff
@@ -5,116 +5,187 @@ local ui = require 'pigui'
-			end
-		end
+	if hypercloud_level == 0 or not target or not target:IsHyperspaceCloud() then
+		return
 	end
 
-	if hypercloud_level > 0 then
-		local target = player:GetNavTarget()
-
-		if target and target:IsHyperspaceCloud() then
-			local arrival = target:IsArrival()
-			local ship = target:GetShip()
-			if ship then
-				local mass = ship.staticMass
-				local path,destName = ship:GetHyperspaceDestination()
-				local date = target:GetDueDate()
-				local dueDate = ui.Format.Datetime(date)
-				local uiPos = shipInfoLowerBound + Vector2(0, 15)
-				local name = (arrival and lc.HYPERSPACE_ARRIVAL_CLOUD or lc.HYPERSPACE_DEPARTURE_CLOUD)
-				local r = ui.addStyledText(uiPos, ui.anchor.right, ui.anchor.top, name , colors.frame, pionillium.medium, nil, colors.lightBlackBackground)
-				uiPos = uiPos + Vector2(0, r.y + offset)
-				r = ui.addFancyText(uiPos, ui.anchor.right, ui.anchor.top, {
-					{ text=lui.HUD_MASS .. ' ', color=colors.reticuleCircleDark, font=pionillium.medium },
-					{ text=mass, color=colors.reticuleCircle, font=pionillium.medium },
-					{ text=lc.UNIT_TONNES, color=colors.reticuleCircleDark, font=pionillium.medium }
-				}, colors.lightBlackBackground)
-				uiPos.y = uiPos.y + r.y + offset
-				r = ui.addStyledText(uiPos, ui.anchor.right, ui.anchor.top, destName, colors.frame, pionillium.medium, nil, colors.lightBlackBackground)
-				uiPos.y = uiPos.y + r.y + offset
-				ui.addStyledText(uiPos, ui.anchor.right, ui.anchor.top, dueDate, colors.frame, pionillium.medium, nil, colors.lightBlackBackground)
-			else
-				local uiPos = Vector2(ui.screenWidth - 30, 1 * ui.gauge_height)
-				ui.addStyledText(uiPos, ui.anchor.right, ui.anchor.top, lc.HYPERSPACE_ARRIVAL_CLOUD_REMNANT , colors.frame, pionillium.medium, nil, colors.lightBlackBackground)
-			end
+	local arrival = target:IsArrival()
+	local ship = target:GetShip()
+
+	local height = ui.getTextLineHeightWithSpacing(font_heading)
+		+ ui.getTextLineHeightWithSpacing() * 3
+		+ ui.getItemSpacing().y
+	local pos, size = ui.rectcut(min, max, height, ui.sides.bottom)
+
+	ui.setNextWindowPos(pos, "Always")
+	ui.setNextWindowSize(size, "Always")
+	ui.setNextWindowPadding(Vector2(0, 0))
+	ui.window("CloudScanner", scannerWindowFlags, function()
+		ui.withFont(font_heading, function()
+			ui.text(target:GetLabel())
+		end)
+		if ship then
+			-- NOTE: ships in arrival clouds have their destination set to the
+			-- source system. This should probably get refactored some time in
+			-- the future. While clever reuse of existing state, it makes for
+			-- challenging maintenance and understanding of the code.
+			local _,systemName = ship:GetHyperspaceDestination()
+			local systemLabel = arrival and lui.HUD_HYPERSPACE_ORIGIN or lui.HYPERSPACE_DESTINATION
+
+			local data = {
+				{ name = lui.HUD_MASS, value = formatMass(ship.staticMass) },
+				{ name = systemLabel, value = systemName },
+				{ name = lui.DATE, value = ui.Format.Datetime(target:GetDueDate()) }
```

Comment:
It is unclear to the user whether this is meant to be the arrival or destination date. A new string is ideally required.

Reply:
Fixed in c157c5e32758ddda2fc1b3442909f89c90f88d1a